src/audio/pipeline/pipeline-params.c

```diff
@@ -170,10 +175,10 @@ static int pipeline_comp_hw_params(struct comp_dev *current,
 
 	/* set buffer parameters */
 	if (calling_buf) {
-		buffer_lock(calling_buf, &flags);
+		calling_buf = buffer_acquire(calling_buf);
 		buffer_set_params(calling_buf, &ppl_data->params->params,
 				  BUFFER_UPDATE_IF_UNSET);
-		buffer_unlock(calling_buf, flags);
+		calling_buf = buffer_release(calling_buf);
```

Comment:
I was thinking about making sure we don't have nested `buffer_acquire()` calls. So, maybe everywhere where a buffer is passed as a parameter it must be acquired by the caller?

Reply:
nested will assert since the ptr alias being passed in will change..